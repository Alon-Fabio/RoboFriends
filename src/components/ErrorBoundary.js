import React, {Component} from 'react'

class hesErrorBoundary extends Component {
    constructor(props){
        super(props);
        this.state = {
            hesError: false
        }
    }

    componentDidCatch(error, info){
        this.setState({hesError:true})
    }
    render(){
        if (this.state.hesError){
            return <h1>Opsss.. this souldnt have happend</h1>
        } 
            
                
        return this.props.children
                
    
    }
}

export default hesErrorBoundary;