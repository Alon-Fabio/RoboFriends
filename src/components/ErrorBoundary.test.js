import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';
import {shallow, mount} from 'enzyme';
 
let wrapper
let wrapperShallow
let wrapperMountWithError
beforeEach(()=>{
    const TestComponent = () =>{
        return <div>"Testing"</div>
    }

    const ComponentWithError = () =>{
        const error = new Error('This is a test error!');
        return <h1>{error}</h1>
    }

    wrapper = renderer.create(
        <ErrorBoundary > 
            <TestComponent />
        </ErrorBoundary>    
    ).toJSON()

    wrapperShallow = shallow(
        <ErrorBoundary > 
            <TestComponent />
        </ErrorBoundary> 
    )

    wrapperMountWithError = mount(
        <ErrorBoundary > 
            <ComponentWithError />
        </ErrorBoundary >    
    )
})

describe("Testing ErroeBoundary", ()=>{
    it("Snapshot with TestComponent", ()=>{
        expect(wrapper).toMatchSnapshot()
    })

    it("Error in component", ()=>{
        expect(wrapperMountWithError.find('h1').text()).toEqual("Opsss.. this souldnt have happend")
    })
})
