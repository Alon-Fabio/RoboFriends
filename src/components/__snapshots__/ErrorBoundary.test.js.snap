// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ErroeBoundary Snapshot with TestComponent 1`] = `
<div>
  "Testing"
</div>
`;
